### DISCUSSION TITLE
Question:  Can you provide some examples of using asynchronous multithreading? I find it very troublesome to handle the lifecycle markers of the Database object. I need to use it in the tokio asynchronous runtime, but I don't know how to deal with these lifecycle tags.

### DISCUSSION THREAD

**Discussion by paulomartins666** [vincent-herlemont/native_db#154] [Q&A]

**Comment by vincent-herlemont** [comment]
@paulomartins666 you can see here an example of `native_db` using `tokio` [tests/watch_tokio.rs](https://github.com/vincent-herlemont/native_db/blob/887e6dd06be1ea80848a5700c540cb0d4530204d/tests/watch_tokio.rs).

>  find it very troublesome to handle the lifecycle markers of the Database object. I need to use it in the tokio asynchronous runtime, but I don't know how to deal with these lifecycle tags.

I don't understand what do you mean with "lifecycle markers", "lifecycle tags".
